### PR TITLE
Add Emacs as an editor for flutter

### PR DESCRIPTION
**Pull request by ericdallo**
Emacs has very good support for running flutter apps via [`lsp-dart`](https://github.com/emacs-lsp/lsp-dart).

### CODE REVIEW

**Comment by ericdallo** [comment]
@googlebot I signed it!

**Comment by sfshaza2** [comment]
@kwalrath, this PR requires staging, as he wants to add a tab for Emacs. Also, I know you've been dealing with some Emacs updates on dart.dev.  :D

Thanks for reviewing!!!

File: src/docs/get-started/editor.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ even better experience. These plugins provide you with code completion, syntax
 highlighting, widget editing assists, run & debug support, and more.
 
 Follow the steps below to add an editor plugin for Android Studio,
-IntelliJ, or VS Code. If you want to use a different editor,
+IntelliJ, VS Code or Emacs. If you want to use a different editor,
```
Comment (kwalrath) [inline src/docs/get-started/editor.md:19]:
```suggestion
IntelliJ, VS Code, or Emacs. If you want to use a different editor,
```

File: src/docs/get-started/editor.md [hunk 2]
```diff
@@ -81,7 +84,19 @@ VS Code is a lightweight editor with Flutter app execution and debug support.
...
+
+## Install Emacs
+
+Emacs is a lightweight editor with support for Flutter and Dart.
 
+* [Emacs][], latest stable version
+
+## Install the LSP dart package
+
+Check [lsp-dart][] for information on how to install and setup.
```
Comment (kwalrath) [inline src/docs/get-started/editor.md:97]: We reverse the sense of sentences like this, per https://developers.google.com/style/clause-order. Maybe:

For information on how to install and use the package, see the 
[lsp-dart documentation][].

It seems to me that the link should be to https://emacs-lsp.github.io/lsp-dart/?

File: src/docs/get-started/editor.md [hunk 3]
```diff
@@ -81,7 +84,19 @@ VS Code is a lightweight editor with Flutter app execution and debug support.
...
 </div>
+<div class="tab-pane" id="vscode" role="tabpanel" aria-labelledby="vscode-tab" markdown="1">
+
+## Install Emacs
+
+Emacs is a lightweight editor with support for Flutter and Dart.
 
+* [Emacs][], latest stable version
+
+## Install the LSP dart package
```
Comment (kwalrath) [inline src/docs/get-started/editor.md:95]:
```suggestion
## Install the lsp-dart package
```

File: src/docs/get-started/editor.md [hunk 4]
```diff
@@ -81,7 +84,19 @@ VS Code is a lightweight editor with Flutter app execution and debug support.
  1. Review the output in the **OUTPUT** pane for any issues.
 
 </div>
+<div class="tab-pane" id="vscode" role="tabpanel" aria-labelledby="vscode-tab" markdown="1">
```
Comment (kwalrath) [inline src/docs/get-started/editor.md:87]:
```suggestion
<div class="tab-pane" id="emacs" role="tabpanel" aria-labelledby="emacs-tab" markdown="1">
```

File: src/docs/get-started/editor.md (hunk 1 above)
Comment (ericdallo) [reply src/docs/get-started/editor.md:19]: Are you sure this colon should exist? I think that in a sentence like that the last prefix to a noun should be an `or`, am I wrong?

File: src/docs/get-started/editor.md (hunk 4 above)
Comment (ericdallo) [reply src/docs/get-started/editor.md:87]: sorry, forgot to replace here

File: src/docs/get-started/editor.md (hunk 1 above)
Comment (kwalrath) [reply src/docs/get-started/editor.md:19]: I'm not sure what you're referring to, but this seems like a case for serial commas: https://developers.google.com/style/commas

**Comment by ericdallo** [comment]
Done @kwalrath!

File: src/docs/get-started/editor.md (hunk 1 above)
Comment (ericdallo) [reply src/docs/get-started/editor.md:19]: Oh, that's new to me, sorry

Comment (ericdallo) [reply src/docs/get-started/editor.md]: Done

**Comment by kwalrath** [comment]
Looking good! I've staged it here:

https://kw-flutter-1.firebaseapp.com/docs/get-started/editor?tab=emacs

@sfshaza2 should @Hixie or someone review this? For history on the dart.dev version of an Emacs page, see https://github.com/dart-lang/site-www/pull/2320. I don't think we need to say too much here, since it's a tab instead of a whole page. But we need to make sure people can find what they need.

**Comment by sfshaza2** [comment]
@Hixie, could you give this a quick once over? The writers, being graduates of UC Berkeley, are quite familiar with vi, but not emacs.  :D

**Comment by sfshaza2** [comment]
OK, Ian hasn't had a chance to look at it, and I don't know emacs, but it looks like a great starting place to me. We can accept PRs to improve/fill out the info as time goes on. Thanks for reviewing and staging the PR, @kwalrath!

**Comment by ericdallo** [comment]
Thanks @sfshaza2, I'll try to open more PR's adding Emacs related info

**Pull request merged**